mttl/models/library/library_transforms.py

```diff
@@ -211,6 +211,131 @@ class WudiMergeConfig(LibraryTransformConfig):
     lr: float = 1e-5
 
 
+@LibraryTransform.register("wudi_merge_after", WudiMergeConfig)
+class WudiMergeAfter(LibraryTransform):
+    """
+    implement the wudimerge in the paper https://arxiv.org/pdf/2503.08099v1
+
+    we multiply the lora A and lora B and then merge the experts to the model(merge after).
+    """
+
+    def __init__(self, config: WudiMergeConfig = None):
+        super().__init__(config or WudiMergeConfig())
+
+    def _get_task_vectors(self, expert):
+        task_vectors = {}
+        for key in expert.expert_weights.keys():
+            base_layer_name = key.split(".lora_")[
+                0
+            ]  # Get base layer name by removing .lora_a or .lora_b
+            if base_layer_name not in task_vectors:
+                task_vectors[base_layer_name] = None
+
+        for layer in task_vectors.keys():
+            lora_a = expert.expert_weights[f"{layer}.lora_a"]
+            lora_b = expert.expert_weights[f"{layer}.lora_b"]
+            task_vectors[layer] = lora_a.data @ lora_b.data
+
+        return task_vectors
+
+    def get_redundant_task_vector(self, layer_name, task_vectors, iter, lr):
```

Comment:
why "redundant"? add docstring too and return types :)

Reply:
This function obtains the optimized merged weights. change the name